identity_core/tests/authentication.rs

```diff
@@ -0,0 +1,546 @@
+use identity_core::{
+    document::DIDDocument,
+    utils::{Authentication, Context, PublicKey, Subject},
+};
+
+use std::str::FromStr;
+
+#[test]
+fn test_auth() {
+    let json_str = r#"
+    {
+        "@context": [
+            "https://w3id.org/did/v1",
+            "https://w3id.org/security/v1"
+        ],
+        "id": "did:iota:123456789abcdefghi",
+        "publicKey": [
+            {
+                "id": "did:iota:123456789abcdefghi#keys-1",
+                "type": "RsaVerificationKey2018",
+                "controller": "did:iota:123456789abcdefghi",
+                "publicKeyPem": "-----BEGIN PUBLIC KEY...END PUBLIC KEY-----"
+            },
+            {
+                "id": "did:iota:123456789abcdefghi#keys-2",
+                "type": "Ed25519VerificationKey2018",
+                "controller": "did:iota:pqrstuvwxyz0987654321",
+                "publicKeyBase58": "H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV"
+            }
+        ],
+        "authentication": [
+            "did:iota:123456789abcdefghi#keys-1",
+            "did:iota:123456789abcdefghi#biometric-1",
+            {
+                "id": "did:iota:123456789abcdefghi#keys-2",
+                "type": "Ed25519VerificationKey2018",
+                "controller": "did:iota:123456789abcdefghi",
+                "publicKeyBase58": "H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV"
+            }
+        ]
+    }
+    "#;
+
+    let doc_1 = DIDDocument::from_str(json_str).unwrap();
+
+    let mut doc_2 = DIDDocument {
+        context: Context::new(vec![
+            "https://w3id.org/did/v1".into(),
+            "https://w3id.org/security/v1".into(),
+        ]),
+        id: Subject::from("did:iota:123456789abcdefghi"),
+        ..Default::default()
+    };
+
+    let key1 = PublicKey::new(
+        "did:iota:123456789abcdefghi#keys-1".into(),
+        "RsaVerificationKey2018".into(),
+        "did:iota:123456789abcdefghi".into(),
+        "publicKeyPem".into(),
+        "-----BEGIN PUBLIC KEY...END PUBLIC KEY-----".into(),
+    )
+    .unwrap();
+
+    let key2 = PublicKey::new(
+        "did:iota:123456789abcdefghi#keys-2".into(),
+        "Ed25519VerificationKey2018".into(),
+        "did:iota:pqrstuvwxyz0987654321".into(),
+        "publicKeyBase58".into(),
```

Comment:
maybe an easier API for this is having `KeyEncodingType` store the actual key data. thoughts?

```rust
#[derive(Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
enum KeyEncodingType {
  PublicKeyBase58(String),
  PublicKeyPem(String),
  ...
}
```

It makes the serialization a bit easier too, eg. `publicKeyJwk` is a map and can be auto-magically handled by serde

Reply:
Have considered it especially since the jwt format has multiple fields associated with it.  For now I just wanted something minimal but will likely be making those changes later.

Reply:
Went ahead and did it and later when the common Object is merged, Ill wrap the JWT in it.